app/models/provider.rb

```diff
@@ -152,7 +152,11 @@ def needs_geolocation?
   end
 
   def full_address
-    [provider_name, address1, address2, address3, address4, postcode].compact.join(", ")
+    address = [provider_name, address1, address2, address3, address4, postcode]
+
+    return nil if address.all?(&:blank?)
```

Comment:
NullObject pattern candidate?

Reply:
I think because we would be returning a string if there was an address the null object would sort of be an empty string. The caller would then check `full_address.blank?` for null-ish-ness.